src/main/scala/viper/gobra/frontend/Config.scala

```diff
@@ -387,38 +389,47 @@ trait PackageAndRecursiveModeConfig extends RawConfig {
 
 case class PackageModeConfig(projectRoot: Path = ConfigDefaults.DefaultProjectRoot.toPath,
                              inputDirectories: Vector[Path], baseConfig: BaseConfig) extends PackageAndRecursiveModeConfig {
-  override lazy val config: Either[String, Config] = {
+  override lazy val config: Either[Vector[VerifierError], Config] = {
     val (errors, mappings) = inputDirectories.map { directory =>
       val sources = getSources(directory, recursive = false, onlyFilesWithHeader = baseConfig.onlyFilesWithHeader)
       // we do not check whether the provided files all belong to the same package
       // instead, we trust the programmer that she knows what she's doing.
       // If they do not belong to the same package, Gobra will report an error after parsing.
       // we simply use the first source's package info to create a single map entry:
-      if (sources.isEmpty) Left(s"no sources found in directory ${directory}")
-      else Right((getPackageInfo(sources.head, projectRoot), sources))
+      if (sources.isEmpty) Left(Vector(ConfigError(s"no sources found in directory $directory")))
+      else for {
+        pkgInfo <- getPackageInfo(sources.head, projectRoot)
+      } yield pkgInfo -> sources
     }.partitionMap(identity)
-    if (errors.length == 1) Left(errors.head)
-    else if (errors.nonEmpty) Left(s"multiple errors have been found while localizing sources: ${errors.mkString(", ")}")
-    else Right(createConfig(mappings.toMap))
+    for {
+      mappings <- if (errors.nonEmpty) Left(errors.flatten) else Right(mappings)
+    } yield createConfig(mappings.toMap)
```

Comment:
nitpick: I think this piece of code becomes easier to read like this:
```suggestion
    if (errors.nonEmpty) {
        Left(errors.flatten)
    } else {
        Right(createConfig(mappings.toMap))
    }
```

Reply:
I agree, thanks!